cmd/sriov/main.go

```diff
@@ -176,30 +171,30 @@ func cmdDel(args *skel.CmdArgs) error {
 
 	sm := sriov.NewSriovManager()
 
-	if !netConf.DPDKMode {
-		if netConf.IPAM.Type != "" {
-			err = ipam.ExecDel(netConf.IPAM.Type, args.StdinData)
-			if err != nil {
-				return err
-			}
-		}
-
-		netns, err := ns.GetNS(args.Netns)
+	if netConf.IPAM.Type != "" {
+		err = ipam.ExecDel(netConf.IPAM.Type, args.StdinData)
 		if err != nil {
-			// according to:
-			// https://github.com/kubernetes/kubernetes/issues/43014#issuecomment-287164444
-			// if provided path does not exist (e.x. when node was restarted)
-			// plugin should silently return with success after releasing
-			// IPAM resources
-			_, ok := err.(ns.NSPathNotExistErr)
-			if ok {
-				return nil
-			}
+			return err
+		}
+	}
 
-			return fmt.Errorf("failed to open netns %s: %q", netns, err)
+	netns, err := ns.GetNS(args.Netns)
```

Comment:
to avoid opening and closing ns fd , as its not being used in dpdk mode, you could move L181-196 into the if block at L197

Reply:
Done